crates/cheatcodes/src/inspector.rs

```diff
@@ -1163,15 +1166,22 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
             !call.is_static;
         if should_check_emits {
             // Not all emits were matched.
-            if self.expected_emits.iter().any(|expected| !expected.found) {
+            if let Some(not_found) = self.expected_emits.iter().find(|expected| !expected.found) {
                 outcome.result.result = InstructionResult::Revert;
-                outcome.result.output = "log != expected log".abi_encode().into();
+                // Where the revert is set and where color mode might be
+                // indicated for a given event that wasn't matched
+                outcome.result.output = Vm::UnemittedEventError {
+                    positionExpected: not_found.sequence + self.expected_emits_offset,
```

Comment:
I think this is incorrect as `sequence` only knows about events which came after `expectEmit` which might not always be the only events emitted during execution

I think we'd want to directly use tracing inspector indexes here, which should be possible after https://github.com/foundry-rs/foundry/pull/8696

Reply:
hey @topocount #8696 just got merged and `CheatcodesExecutor` was extended with `tracing_inspector` fn which provides a reference to a tracing inspector from which you can check the log indicies directly

that way `sequence` value can be just set to something like `tracer.traces().nodes().last().expect("no trace nodes").logs.len()`

Reply:
I should have time to work on this this week

Reply:
@klkvr I've added `tracer.traces().nodes().last().expect("no trace nodes").logs.len()` to `expect_emit` but still need the offset to cover an edge case where a prior call completes successfully, otherwise that prior call is erroneously highlighted red instead. 

`not_found.sequence` holds the tracer length call from from when the expected emit payload is initially pushed onto the stack.

Reply:
hmm there are still edge cases that aren't satisfied by this, namely if multiple expectedEmits are added for a given call, the incorrect one (the one that is matched) is highlighted. I need to dig into this more.

Reply:
ah i think this is the root of the issue. events are matched backwards
https://github.com/topocount/foundry/blob/2d9590e72b9883f34be31e5710ca7cabc27dc132/crates/cheatcodes/src/test/expect.rs#L594-L599

Reply:
it doesn't look like this is incrementing. Is this array lazily constructed then? 

> that way sequence value can be just set to something like tracer.traces().nodes().last().expect("no trace nodes").logs.len()

Reply:
Here's a debug trace where were expecting the same event twice:
<details>
<summary> Click here </summary>

```
[crates/cheatcodes/src/inspector.rs:1366:17] &self.expected_emits = [
    ExpectedEmit {
        depth: 1,
        log: Some(
            LogData {
                topics: [
                    0xf6ef72180c46cadbda80997bfa03fc39b76911c9bc988da15e4a47d55d687a31,
                ],
                data: 0x00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000005,
            },
        ),
        checks: [
            true,
            false,
            false,
            false,
            true,
        ],
        address: Some(
            0x5615deb798bb3e4dfa0139dfa1b3d433cc23b72f,
        ),
        anonymous: false,
        found: false,
        sequence: 0,
    },
    ExpectedEmit {
        depth: 1,
        log: Some(
            LogData {
                topics: [
                    0xf6ef72180c46cadbda80997bfa03fc39b76911c9bc988da15e4a47d55d687a31,
                ],
                data: 0x00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000005,
            },
        ),
        checks: [
            true,
            false,
            false,
            false,
            true,
        ],
        address: Some(
            0x5615deb798bb3e4dfa0139dfa1b3d433cc23b72f,
        ),
        anonymous: false,
        found: true,
        sequence: 0,
    },
]
```
</details>